doc/contribute/external.rst

```diff
@@ -103,6 +103,32 @@ With this approach the code is considered as being developed externally, and
 thus it is not automatically subject to the requirements of the previous
 section.
 
+Integration in main manifest file (west.yaml)
+++++++++++++++++++++++++++++++++++++++++++++++
+
+Integrating external code into the main :file:`west.yml` manifest file is
+limited to code that is used by a Zephyr subsystem (libraries) or by a platform
+or drivers (HAL) or tooling needed to test or build Zephyr components.
+
+Integration as optional modules
++++++++++++++++++++++++++++++++++++++++++
+
+Standalone or loose integration of modules/projects without any incoming
+dependencies shall be made optional and shall be kept standalone. Optional
+projects that provide value to users directly and through a Zephyr subsystem or
+platform shall be added to an optional manifest file that is filtered by
+default. (:file:`submanifests/optional.yml`).
+
+Such optional projects might include samples and tests in their own repositories.
```

Comment:
Similar thing here, but add language that the user should check the release notes of the module for compatibility with a specific Zephyr release.

Reply:
Should stand-alone repos be manifest-modules?

Reply:
no, I do not think there is no needed for that and it does not add any value, simply `zephyr/module.yaml`